BigBaseV2/src/hooks/protections/received_clone_sync.cpp

```diff
@@ -20,21 +35,317 @@ namespace big
+	{
+		if (node->IsParentNode())
+		{
+			for (auto child = node->m_first_child; child; child = child->m_next_sibling) {
+				CPhysicalAttachDataNode* attach_node = get_attach_node_from_object(child);
+				if(attach_node != nullptr)
+					return attach_node;
+			}
+		}
+		else if (node->IsDataNode())
+		{
+			if (typeid(*node).hash_code() == 0xE6B80DD4C167CFF) //CPhysicalAttachDataNode
+				return dynamic_cast<CPhysicalAttachDataNode*>(node);
+		}
+		return nullptr;
+	}
+
+	CPedAttachDataNode* get_ped_attach_node_from_object(rage::netSyncNodeBase* node)
+	{
+		if (node->IsParentNode())
+		{
+			for (auto child = node->m_first_child; child; child = child->m_next_sibling) {
+				CPedAttachDataNode* attach_node = get_ped_attach_node_from_object(child);
+				if (attach_node != nullptr)
+					return attach_node;
+			}
+		}
+		else if (node->IsDataNode())
+		{
+			if (typeid(*node).hash_code() == 0xD3A0FBD0D1CFD23A) //CPedAttachDataNode
+				return dynamic_cast<CPedAttachDataNode*>(node);
+		}
+		return nullptr;
+	}
+
+	bool is_attachment_infinite(CPhysicalAttachDataNode* node, uint16_t object_id) {
+		if (rage::netObject* attached_object = (*g_pointers->m_network_object_mgr)->find_object_by_id(node->m_attached_to, false); attached_object)
+		{
+			if (rage::netSyncTree* tree = attached_object->GetSyncTree(); tree)
+			{
+				if (rage::netSyncNodeBase* base_node = tree->m_sync_node; base_node) {
+
+					CPhysicalAttachDataNode* attached_attach_node = get_attach_node_from_object(base_node);
+					if (attached_attach_node && attached_attach_node->m_attached)
+					{
+						if (attached_attach_node->m_attached_to == object_id) {
+							return true;
+						}
+						else {
+							return is_attachment_infinite(attached_attach_node, object_id);
+						}
+					}
+
+				}
+			}
+		}
+
+		return false;
+	}
+
+	bool is_ped_attachment_infinite(CPedAttachDataNode* node, uint16_t object_id) {
+		if (rage::netObject* attached_object = (*g_pointers->m_network_object_mgr)->find_object_by_id(node->m_attached_to, false); attached_object)
+		{
+			if (rage::netSyncTree* tree = attached_object->GetSyncTree(); tree)
+			{
+				if (rage::netSyncNodeBase* base_node = tree->m_sync_node; base_node) {
+
+					CPedAttachDataNode* attached_attach_node = get_ped_attach_node_from_object(base_node);
+					if (attached_attach_node && attached_attach_node->m_attached)
+					{
+						if (attached_attach_node->m_attached_to == object_id) {
+							return true;
+						}
+						else {
+							return is_ped_attachment_infinite(attached_attach_node, object_id);
+						}
+					}
+
+				}
+			}
+		}
+
+		return false;
+	}
+
+	bool check_node(rage::netSyncNodeBase* node, CNetGamePlayer* sender, uint16_t object_id)
+	{
+		if (node->IsParentNode())
+		{
+			bool block = false;
+			for (auto child = node->m_first_child; child; child = child->m_next_sibling) {
+				if (check_node(child, sender, object_id))
+					block = true;
```

Comment:
return true in the for loop immediately? I'm not familiar but it seems like a waste of checking the other nodes if we're going to block it anyways.